conans/client/rest/rest_client_common.py

```diff
@@ -202,25 +204,25 @@ def remove_packages(self, conan_reference, package_ids=None):
         """ Remove any packages specified by package_ids"""
         self.check_credentials()
         payload = {"package_ids": package_ids}
-        url = "%s/conans/%s/packages/delete" % (self.remote_api_url, '/'.join(conan_reference))
+        url = self._recipe_url(conan_reference) + "/packages/delete"
         return self._post_json(url, payload)
 
     @handle_return_deserializer()
     def _remove_conanfile_files(self, conan_reference, files):
         """ Remove recipe files """
         self.check_credentials()
         payload = {"files": [filename.replace("\\", "/") for filename in files]}
-        url = "%s/conans/%s/remove_files" % (self.remote_api_url, '/'.join(conan_reference))
+        url = self._recipe_url(conan_reference)
+        url = "/%s/remove_files" % url
         return self._post_json(url, payload)
 
     @handle_return_deserializer()
     def _remove_package_files(self, package_reference, files):
         """ Remove package files """
         self.check_credentials()
         payload = {"files": [filename.replace("\\", "/") for filename in files]}
-        url = "%s/conans/%s/packages/%s/remove_files" % (self.remote_api_url,
-                                                         "/".join(package_reference.conan),
-                                                         package_reference.package_id)
+        url = self._package_url(package_reference)
+        url = "/%s/remove_files" % (url, package_reference.package_id)
```

Comment:
Possible bug here.

Reply:
True. This code is almost never execute. It is used in the upload, when we perform a diff of the files to be uploaded, in case some file has been deleted, but as we always upload the same files to a package... (zipped contents...). Should we remove the diff for packages to simplify?

Reply:
This was done because sometimes errors while uploading the conaninfo.txt after 30mins uploading a huge conan_package.tgz was annoying for some users that wanted to retry and have it fast the second time.

Reply:
Anyway, it should be done in a different PR to not increase this.

Reply:
In any case, it seems this should be covered by some test, but it isn't. Is it worth to add it (directly to develop)?

Reply:
Oh, now I see. The code is Ok (but weird, I've simplified) and probably it is covered by a test somewhere.